src/app/screens/Settings.tsx

```diff
@@ -133,8 +133,18 @@ function Settings() {
               label={t("camera_access.allow")}
               onClick={async () => {
                 try {
-                  await Html5Qrcode.getCameras();
-                  setCameraPermissionsGranted(true);
+                  const devices = await QrScanner.listCameras();
+                  const video =
+                    document.querySelector<HTMLVideoElement>("#reader");
+                  if (video && devices && devices.length) {
```

Comment:
```suggestion
                  if (video && devices && devices.length > 0) {
```

I know that `0` is falsey in JS but imo that's a language quirk and I find the code more readable for humans if we don't rely on it.